nncf/experimental/tensor/functions/torch_numeric.py

```diff
@@ -299,3 +330,41 @@ def _(a: torch.Tensor, axis: int = -1, descending=False, stable=False) -> torch.
 @numeric.diag.register(torch.Tensor)
 def _(a: torch.Tensor, k: int = 0) -> torch.Tensor:
     return torch.diag(a, diagonal=k)
+
+
+@numeric.logical_or.register(torch.Tensor)
+def _(x1: torch.Tensor, x2: torch.Tensor) -> torch.Tensor:
+    return torch.logical_or(x1, x2)
+
+
+@numeric.zero_elements.register(torch.Tensor)
+def zero_elements(x: torch.Tensor) -> torch.Tensor:
+    return torch.abs(x) < torch.finfo(x.dtype).eps
+
+
+@numeric.masked_mean.register(torch.Tensor)
+def _(
+    x: torch.Tensor, mask: Optional[torch.Tensor], axis: Union[int, Tuple[int, ...], List[int]], keepdims=False
+) -> torch.Tensor:
+    if mask is None:
+        return torch.mean(x, axis=axis, keepdims=keepdims)
+    device = x.device
+    masked_x = np.ma.array(x.detach().cpu().numpy(), mask=mask.detach().cpu().numpy())
+    result = np.ma.mean(masked_x, axis=axis, keepdims=keepdims).astype(masked_x.dtype)
+    if isinstance(result, np.ma.MaskedArray):
+        result = result.data
+    return torch.tensor(result).to(device=device)
+
+
+@numeric.masked_median.register(torch.Tensor)
+def _(
+    x: torch.Tensor, mask: Optional[torch.Tensor], axis: Union[int, Tuple[int, ...], List[int]], keepdims=False
+) -> torch.Tensor:
+    if mask is None:
+        return torch.median(x, axis=axis, keepdims=keepdims)
+    device = x.device
+    masked_x = np.ma.array(x.detach().cpu().numpy(), mask=mask.detach().cpu().numpy())
```

Comment:
It looks like not optimal implementation for torch backend in practice  for gpu. Could you think about more optimal implementation without copying tensor in gpu case?

Reply:
https://github.com/openvinotoolkit/nncf/blob/24f0ee707cd5a35abf351f8647efc0fbd0ee273b/nncf/torch/tensor_statistics/collectors.py#L103-L114

Reply:
reworked by torch.nanmedian